packages/twenty-front/src/modules/apollo/utils/getImpersonationTokensFromStorage.ts.ts

```diff
@@ -0,0 +1,47 @@
+import { isDefined, parseJson } from 'twenty-shared/utils';
+import { type AuthTokenPair } from '~/generated/graphql';
+import { cookieStorage } from '~/utils/cookie-storage';
+
+export const getImpersonationTokensFromStorage = (): AuthTokenPair | undefined => {
+  const raw = cookieStorage.getItem('impersonationTokenPair');
+  if (!isDefined(raw)) {
+    return;
+  }
+
+  const authTokenPair = parseJson<Partial<AuthTokenPair>>(raw);
+  if (!isDefined(authTokenPair)) {
+    cookieStorage.removeItem('impersonationTokenPair');
```

Comment:
This feels like a code smell. The function is called getXXX (suggests it's a query) but then removes stuff within it (mutates data)